gradle-jdks/src/test/groovy/com/palantir/gradle/jdks/GradleJdkPatcherIntegrationTest.groovy

```diff
@@ -0,0 +1,229 @@
+/*
+ * (c) Copyright 2024 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.gradle.jdks
+
+import com.google.common.base.Splitter
+import com.google.common.collect.Iterables
+import com.google.common.collect.Sets
+import com.palantir.gradle.jdks.setup.CommandRunner
+import nebula.test.IntegrationSpec
+import spock.lang.TempDir
+
+import java.nio.file.Files
+import java.nio.file.Path
+
+class GradleJdkPatcherIntegrationTest extends IntegrationSpec {
+
+    private static String GRADLE_7VERSION = "7.6.2";
+    private static String GRADLE_8VERSION = "8.4";
+    private static AmazonCorrettoJdkDistribution CORRETTO_JDK_DISTRIBUTION = new AmazonCorrettoJdkDistribution();
+    private static String CORRETTO_DISTRIBUTION_URL_ENV = "CORRETTO_DISTRIBUTION_URL";
```

Comment:
For these things that are just used in one place, I'd just inline them - not sure making them static adds much.

Reply:
Perhaps the serial numbers make sense, as they're just a string of numbers and need a name.